kernel/thread.c

```diff
@@ -657,6 +727,10 @@ k_tid_t z_impl_k_thread_create(struct k_thread *new_thread,
 {
 	__ASSERT(!arch_is_in_isr(), "Threads may not be created in ISRs");
 
+#ifdef CONFIG_VALIDATE_THREAD_STACK_POINTER
+	z_check_thread_stack(stack, stack_size, options);
+#endif
```

Comment:
This is a little like tautology (this pointer is within the memory range that the pointer is in).

It probably also gives a false sense of security, since stack overflows occur after thread creation, since a check would need to be done whenever a thread was scheduled.

However, we already have both hardware stack protection and stack canaries (software stack checking).